server/netop_handler.go

```diff
@@ -261,39 +281,41 @@ func (s *NetOps) deleteTcForSlice(sliceID string) error {
 func (s *NetOps) configureTcForSlice(sliceID string, newTc *TcInfo) error {
 	sliceInfo, found := NetOpHandle[sliceID]
 	if !found {
-		errVal := fmt.Sprintf("SliceId %v is not found", sliceID)
+		errVal := SliceIdNotFound(sliceID)
 		return errors.New(errVal)
 	}
 
-	if sliceInfo.tc != nil {
-		// Check if there are any changes in TC parameters.
-		if (*sliceInfo.tc) == (*newTc) {
-			logger.GlobalLogger.Infof("No change in Slice TC params, ignoring update")
-			return nil
-		} else {
-			logger.GlobalLogger.Infof("Slice TC params updated. Old: %v, New: %v", sliceInfo.tc, newTc)
-			// Modify parent class config
-			tcCmd := fmt.Sprintf("tc class replace dev %s parent %d: classid %s htb rate %dkbit burst 64k",
-				netIface, htbRootHandleId, sliceInfo.tcParentClassFqId, newTc.bwCeiling)
-			cmdOut, err := runTcCommand(tcCmd)
-			if err != nil {
-				errStr := fmt.Sprintf("tc Command: %v execution failed with err: %v and stderr : %v", tcCmd, err, cmdOut)
-				logger.GlobalLogger.Errorf(errStr)
-				return errors.New(errStr)
-			}
+	if sliceInfo.tc == nil {
+		logger.GlobalLogger.Infof("Recieved nil value for tc configuration from slice controller")
+		return nil
+	}
 
-			// Modify leaf class config
-			tcCmd = fmt.Sprintf("tc class replace dev %s parent %s classid %s htb rate %dkbit ceil %dkbit burst 32k",
-				netIface, sliceInfo.tcParentClassFqId, sliceInfo.tcLeafClassFqId, newTc.bwGuaranteed, newTc.bwCeiling)
-			cmdOut, err = runTcCommand(tcCmd)
-			if err != nil {
-				errStr := fmt.Sprintf("tc Command: %v execution failed with err: %v and stderr : %v", tcCmd, err, cmdOut)
-				logger.GlobalLogger.Errorf(errStr)
-				return errors.New(errStr)
-			}
-			sliceInfo.tc = newTc
-			return nil
+	// Check if there are any changes in TC parameters.
+	if (*sliceInfo.tc) == (*newTc) {
+		logger.GlobalLogger.Infof("No change in Slice TC params, ignoring update")
+		return nil
+	} else {
```

Comment:
We can simplify this further, move the else block out in normal flow. if block return ensures the rest of code will only run when the condition is false.